Undefined reference to Frame::ToInfoString function when log level < LOG_LEVEL_INFO

Building OpenThread I see the linker fails if the OPENTHREAD_CONFIG_LOG_LEVEL is set to OT_LOG_LEVEL_NOTE.  However, the link succeeds if it is set to LOG_LEVEL_INFO.

The commit ID used for testing is  0d75bbe.

The error message I see shows:
/examples/../src/core/thread/mesh_forwarder.cpp:1774: undefined reference to `ot::Mac::Frame::ToInfoString() const'

For building I've used the efr32mg12 platform and added 
 -DOPENTHREAD_CONFIG_LOG_LEVEL=OT_LOG_LEVEL_NOTE to the COMMONCFLAGS section of the makefile and built with:

$ make -f examples/Makefile-efr32mg12 COMMISSIONER=1 JOINER=1 DHCP6_CLIENT=1 DHCP6_SERVER=1 BOARD=BRD4161A

It looks like Frame::ToInfoString in is only defined in mac_frame.cpp if log level >= OT_LOG_LEVEL_INFO and OPENTHREAD_CONFIG_LOG_MAC is set to 1.

However this function is used in calls to otLogNoteMac macros in mesh_forwarder.cpp.

Should these calls be converted to otLogInfoMac or should the ifdef in mac_frame.cpp be changed to OT_LOG_LEVEL_NOTE?

--- a/src/core/mac/mac_frame.cpp
+++ b/src/core/mac/mac_frame.cpp
@@ -1060,7 +1060,7 @@ uint16_t Frame::GetFcsSize(void) const
 
 // LCOV_EXCL_START
 
-#if (OPENTHREAD_CONFIG_LOG_LEVEL >= OT_LOG_LEVEL_INFO) && (OPENTHREAD_CONFIG_LOG_MAC == 1)
+#if (OPENTHREAD_CONFIG_LOG_LEVEL >= OT_LOG_LEVEL_NOTE) && (OPENTHREAD_CONFIG_LOG_MAC == 1)
 
 Frame::InfoString Frame::ToInfoString(void) const
 {
@@ -1136,7 +1136,7 @@ BeaconPayload::InfoString BeaconPayload::ToInfoString(void) const
                       IsNative() ? "yes" : "no");
 }
 
-#endif // #if (OPENTHREAD_CONFIG_LOG_LEVEL >= OT_LOG_LEVEL_INFO) && (OPENTHREAD_CONFIG_LOG_MAC == 1)
+#endif // #if (OPENTHREAD_CONFIG_LOG_LEVEL >= OT_LOG_LEVEL_NOTE) && (OPENTHREAD_CONFIG_LOG_MAC == 1)
 
 // LCOV_EXCL_STOP
 

